py3status/modules/battery_level.py

```diff
@@ -315,6 +318,16 @@ def _parse_battery_info(sys_path):
             current_now = r.get("POWER_SUPPLY_CURRENT_NOW", 0)
             voltage_now = r.get("POWER_SUPPLY_VOLTAGE_NOW", 0)
 
+            # missing values may indicate this is not a battery and should be skipped
+            if not (
+                capacity
+                and present_rate
```

Comment:
I think this check should not be included here: it avoids to include fully charged batteries that will enter in the `except ZeroDivisionError` on R343, instead.

Reply:
True... Maybe instead of relying on a division error to detect fully charged, we could check for the existence of the corresponding key/value. That would be easier to read. What do you think ?

Reply:
it can be done with a pre-check, before the division: if all other values are presents but only `present_rate` is missing, the battery is present and fully charge (or, at least, not discharging and not charging)

Reply:
I added a proposal in my last commit from PR #1949 (048c26c89fea7ab51ed1130366af0d997d326d13)

```
            # missing values may indicate this is not a battery and should be skipped
            print(path, capacity, remaining_energy, present_rate, current_now, voltage_now)
            if (
                capacity == None
                or present_rate == None
                or remaining_energy == None
                or current_now == None
                or voltage_now == None
            ):
                continue

            battery = {}
            battery["capacity"] = capacity
            battery["charging"] = "Charging" in r["POWER_SUPPLY_STATUS"]
            battery["percent_charged"] = int(
                math.floor(remaining_energy / capacity * 100)
            )
            if present_rate == 0:
                # Battery is either full charged or is not discharging
                battery["time_remaining"] = FULLY_CHARGED
            else:
                if battery["charging"]:
                    time_in_secs = (capacity - remaining_energy) / present_rate * 3600
                else:
                    time_in_secs = remaining_energy / present_rate * 3600
                battery["time_remaining"] = self._seconds_to_hms(time_in_secs)

            battery["power"] = current_now * voltage_now / 1e12
```